### ISSUE TITLE
Question: Why is this patch request INVALID?

### ISSUE THREAD

**Issue by kskumgk63** [elimity-com/scim#105]
Hi, there.

I am trying to integrate My App which uses this library with AzureAD.

I do not know why this Patch request AzureAD sent is invalid.
It seems valid to me.

```
{
       "schemas": [
               "urn:ietf:params:scim:api:messages:2.0:PatchOp"
       ],
       "Operations": [
               {
                       "op": "Replace",
                       "path": "emails[type eq \"work\"].value",
                       "value": "hoge@example.com"
               }
       ]
}
```

This library treats this request as `ScimErrorInvalidSyntax` here.

https://github.com/elimity-com/scim/blob/a9839c78b5161b4f63e5a58ca0ac8c1ac5a5ee80/schema/core.go#L179-L183

I want to be clear if the request AzureAD sent is wrong or this library has bug.
Do you have any ides? @di-wu

**Comment by di-wu** [comment]
Hi @kskumgk63,

Thanks for the many issues! You've found a lot of bug already! 👍🏽 

---

This is a **valid** operation.
The operation you described in words:
> Replace the _value field_ of the complex email attribute which has as type "work" to the given email address.

This is a bug in the library.

To clarify the expected behaviour:
**The brackets (`[]`) select the value record, whereas `.value` selects the sub attribute.**

**Comment by di-wu** [comment]
I don't have time today to look into it further. I will take a look at it _Wednesday_.

**Comment by kskumgk63** [comment]
@di-wu Could you review my PR? #106

**Comment by di-wu** [comment]
Closed by #106
